packages/teuchos/CMakeLists.txt

```diff
@@ -448,9 +446,28 @@ TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_KOKKOS_PROFILING
   "Wrap every Teuchos timer with a Kokkos profiling region."
   "${${PACKAGE_NAME}_KOKKOS_PROFILING_DEFAULT}")
 IF (${PACKAGE_NAME}_KOKKOS_PROFILING)
+  # Enabling Kokkos profiling hooks in Teuchos timers requires that the
+  # Kokkos package be enabled
+  IF(NOT(DEFINED ${PROJECT_NAME}_ENABLE_Kokkos AND ${PROJECT_NAME}_ENABLE_Kokkos))
+    MESSAGE(FATAL_ERROR "Setting Teuchos_KOKKOS_PROFILING=ON requires that the Kokkos package is enabled.")
+  ENDIF()
   MESSAGE(STATUS "Wrapping every Teuchos timer with a Kokkos profiling region.")
 ENDIF()
 
+#Option Teuchos_TIMER_KOKKOS_FENCE
+SET(Teuchos_TIMER_KOKKOS_FENCE_DEFAULT OFF)
+TRIBITS_ADD_OPTION_AND_DEFINE(
+  Teuchos_TIMER_KOKKOS_FENCE
+  HAVE_TEUCHOS_TIMER_KOKKOS_FENCE
+  "Call Kokkos::fence() whenever a Teuchos timer starts or stops."
+  "${Teuchos_KOKKOS_PROFILING_DEFAULT}")
```

Comment:
@brian-kelley I think this should be using `Teuchos_TIMER_KOKKOS_FENCE_DEFAULT` as the default, no?

Reply:
Yes, that was dumb. Good catch. Fixed in #13081